traffic_monitor/health/cache.go

```diff
@@ -53,14 +53,44 @@ func GetVitals(newResult *cache.Result, prevResult *cache.Result, mc *tc.Traffic
 		return
 	}
 
+	if mc == nil {
+		log.Errorf("mc TrafficMonitorConfigMap cannot be nil")
+		return
+	}
+
 	if newResult.InterfaceVitals == nil {
 		newResult.InterfaceVitals = map[string]cache.Vitals{}
 	}
 
 	// proc.loadavg -- we're using the 1 minute average (!?)
 	newResult.Vitals.LoadAvg = newResult.Statistics.Loadavg.One
 
-	for ifaceName, iface := range newResult.Interfaces() {
+	if mc.TrafficServer[newResult.ID].Interfaces == nil {
+		log.Debugf("no interfaces reported in config map to be monitored")
+		return
+	}
+
+	var monitoredInterfaces []tc.ServerInterfaceInfo
+	for _, srvrIfaceInfo := range mc.TrafficServer[newResult.ID].Interfaces {
+		if srvrIfaceInfo.Monitor {
+			monitoredInterfaces = append(monitoredInterfaces, srvrIfaceInfo)
+		}
+	}
+
+	if len(monitoredInterfaces) == 0 {
+		log.Debugf("no interfaces selected to be monitored for %v", newResult.ID)
+		return
+	}
+
+	for _, monitoredInterface := range monitoredInterfaces {
+		ifaceName := monitoredInterface.Name
+		iface, exists := newResult.Interfaces()[ifaceName]
```

Comment:
Maybe add some similar checks here like my last comment^^

Reply:
Since this is a map, the `exists` value will tell us if a match was found and we can log the information thusly. Otherwise the map returns an empty interface, rather than `nil`. This should be satisfactory as written.